modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaJAXRSServerCodegen.java

```diff
@@ -201,9 +201,10 @@ private String implFileFolder(String output) {
         return outputFolder + "/" + output + "/" + apiPackage().replace('.', '/');
     }
 
-    @Override
-    public boolean shouldOverwrite(String filename) {
-        return super.shouldOverwrite(filename) && !filename.endsWith("ServiceImpl.java") && !filename.endsWith("ServiceFactory.java");
-    }
+// TODO: @scannerscan: I think we need to allow overriding because the number of methods and signature can be changed.
```

Comment:
Shall we keep `shouldOverwrite` for the time being before we move to .swagger-codegen-ignore?